File: sdk/sdk-api-java/src/main/java/com/xpay/sdk/api/entity/Request.java
package com.xpay.sdk.api.entity;

import com.xpay.sdk.api.utils.JsonUtil;

import java.util.Map;
import java.util.TreeMap;

/**
 * 商户请求的VO
 */
public class Request {
    private String method;
    private String version;
    private Object data;
    private String randStr;
    private String signType;
    private String mchNo;
    private String secKey;
    private String timestamp;//时间戳，精确倒毫秒

    public String getMethod() {
        return method;
    }

    public void setMethod(String method) {
        this.method = method;
    }

    public String getVersion() {
        return version;
    }

    public void setVersion(String version) {
        this.version = version;
    }

    public Object getData() {
        return data;
    }

    public void setData(Object data) {
        this.data = data;
    }

    public String getRandStr() {
        return randStr;
    }

    public void setRandStr(String randStr) {
        this.randStr = randStr;
    }

    public String getSignType() {
        return signType;
    }

    public void setSignType(String signType) {
        this.signType = signType;
    }

    public String getMchNo() {
        return mchNo;
    }

    public void setMchNo(String mchNo) {
        this.mchNo = mchNo;
    }

    public String getSecKey() {
        return secKey;
    }

    public void setSecKey(String secKey) {
        this.secKey = secKey;
    }

    public String getTimestamp() {
        return timestamp;
    }

    public void setTimestamp(String timestamp) {
        this.timestamp = timestamp;
    }

    public Map<String, String> toSortMap(){
        Map<String, String> map = new TreeMap<>();//按字典序排序(升序)
        map.put("method", method);
        map.put("version", version);
        map.put("randStr", randStr);
        map.put("signType", signType);
        map.put("mchNo", mchNo);
        map.put("timestamp", timestamp);
        map.put("data", data instanceof String ? (String) data : JsonUtil.toJson(data));
        if(secKey != null){
            map.put("secKey", secKey);
        }
        return map;
    }

    public void joinSecKey(String key, String iv){
        this.secKey = key + ":" + iv;
    }
}

File: sdk/sdk-api-php/entity/Request.php
<?php
namespace xpay;

/**
 * http请求参数类
 * Class Request
 * @package entity
 */
class Request implements \JsonSerializable {
    private string $method = '';
    private string $version = "1.0";
    private $data;
    private string $randStr = '';
    private string $signType = "2";
    private string $mchNo = '';
    private string $secKey = '';
    private string $timestamp = '';

    /**
     * 进行json序列化时返回的内容
     * @return array|mixed
     */
    public function jsonSerialize()
    {
        $vars = get_object_vars($this);
        return $vars;
    }

    /**
     * @return string
     */
    public function getMethod(): string
    {
        return $this->method;
    }

    /**
     * @param string $method
     */
    public function setMethod(string $method): void
    {
        $this->method = $method;
    }

    /**
     * @return string
     */
    public function getVersion(): string
    {
        return $this->version;
    }

    /**
     * @param string $version
     */
    public function setVersion(string $version): void
    {
        $this->version = $version;
    }

    /**
     * @return mixed
     */
    public function getData()
    {
        return $this->data;
    }

    /**
     * @param mixed $data
     */
    public function setData($data): void
    {
        $this->data = $data;
    }

    /**
     * @return string
     */
    public function getRandStr(): string
    {
        return $this->randStr;
    }

    /**
     * @param string $randStr
     */
    public function setRandStr(string $randStr): void
    {
        $this->randStr = $randStr;
    }

    /**
     * @return string
     */
    public function getSignType(): string
    {
        return $this->signType;
    }

    /**
     * @param string $signType
     */
    public function setSignType(string $signType): void
    {
        $this->signType = $signType;
    }

    /**
     * @return string
     */
    public function getMchNo(): string
    {
        return $this->mchNo;
    }

    /**
     * @param string $mchNo
     */
    public function setMchNo(string $mchNo): void
    {
        $this->mchNo = $mchNo;
    }

    /**
     * @return string
     */
    public function getSecKey(): string
    {
        return $this->secKey;
    }

    /**
     * @param string $secKey
     */
    public function setSecKey(string $secKey): void
    {
        $this->secKey = $secKey;
    }

    /**
     * @return string
     */
    public function getTimestamp(): string
    {
        return $this->timestamp;
    }

    /**
     * @param string $timestamp
     */
    public function setTimestamp(string $timestamp): void
    {
        $this->timestamp = $timestamp;
    }

    public function joinSecKey(string $secKey, string $iv) : void
    {
        $this->secKey = $secKey . ":" . $iv;
    }
}